app/movie/[imdbid]/page.tsx

```diff
@@ -0,0 +1,62 @@
+"use client";
+
+import * as React from "react";
+import {redirect} from 'next/navigation';
+import { MovieField } from "@/lib/MovieField";
+import {HiMiniHeart,HiOutlineHeart } from "react-icons/hi2";
+import { MoviesContext } from "@/context/movieData.context";
+import movieData from "../../../lib/top_100_movies.json";
+
+interface PropType {
+  params: {
+    imdbid: string;
+  };
+}
+
+const MovieDetail: React.FC<PropType> = ({ params }) => {
+  const [isFavourited, setIsFavourited] = React.useState(false);
+  const {favMovies, addFavMovie, removeFavMovie} = React.useContext(MoviesContext);
+  const movie = movieData.find((movie) => movie.imdbid === params.imdbid);
+  if(movie === undefined) redirect("/");
+  const {
+    title,
+    rating,
+    id,
+    year,
+    image,
+    description,
+    trailer,
+    genre,
+    director,
+    writers,
+    imdbid
+  } = movie as MovieField;
+
+  React.useEffect(() => {
+    const favMovie = favMovies.find((favMovie) => favMovie.imdbid === params.imdbid)
+    if(favMovie) {
+      setIsFavourited(true)
+    } else {
+      setIsFavourited(false)
+    }
+  }, [favMovies, params.imdbid])
+
+  const handleAddFav = (_movie: MovieField) => {
+    addFavMovie(_movie);
+    setIsFavourited(true)
+  }
+  const handleRemoveFav = (_imdbid: string) => {
+    removeFavMovie(_imdbid);
+    setIsFavourited(false)
+  }
+  return( 
+    <div>
+      <h1 className="text-4xl">{title}</h1>
+      <div className="flex items-center gap-2">
+      <span className="text-sm uppercase">saved to wishlist</span>
```

Comment:
todo: I believe the english is incorrect here. When you say "saved" it's in the past, like already done. If it's not done, then you say "save" (as action). So the code should be:

```
<span className="text-sm uppercase">
     {isFavourited ? "saved to wishlist" : "save to wishlist"}
</span>
```